Rubberduck.CodeAnalysis/Inspections/Concrete/ObsoleteTypeHintInspection.cs

```diff
@@ -32,32 +35,100 @@ namespace Rubberduck.Inspections.Concrete
     /// </example>
     public sealed class ObsoleteTypeHintInspection : InspectionBase
     {
+        private readonly IDeclarationFinderProvider _declarationFinderProvider;
+
         public ObsoleteTypeHintInspection(RubberduckParserState state)
-            : base(state) { }
+            : base(state)
+        {
+            _declarationFinderProvider = state;
+        }
 
         protected override IEnumerable<IInspectionResult> DoGetInspectionResults()
         {
-            var results = UserDeclarations.ToList();
-
-            var declarations = from item in results
-                where item.HasTypeHint
-                select
-                    new DeclarationInspectionResult(this,
-                        string.Format(InspectionResults.ObsoleteTypeHintInspection,
-                            InspectionsUI.Inspections_Declaration, item.DeclarationType.ToString().ToLower(),
-                            item.IdentifierName), item);
-
-            var references = from item in results.SelectMany(d => d.References)
-                where item.HasTypeHint()
-                select
-                    new IdentifierReferenceInspectionResult(this,
-                        string.Format(InspectionResults.ObsoleteTypeHintInspection,
-                            InspectionsUI.Inspections_Usage, item.Declaration.DeclarationType.ToString().ToLower(),
-                            item.IdentifierName),
-                        State,
-                        item);
-
-            return declarations.Union<IInspectionResult>(references);
+            var finder = _declarationFinderProvider.DeclarationFinder;
+
+            var results = new List<IInspectionResult>();
+            foreach (var moduleDeclaration in State.DeclarationFinder.UserDeclarations(DeclarationType.Module))
```

Comment:
shouldn't that be `var moduleDeclaration in finder.UserDeclarations(DeclarationType.Module)`?

Reply:
Good catch! I will adjust it later today.